### PR TITLE
NOISSUE - Fix Authorization in Things Service

### PR DESCRIPTION
**Pull request by rodneyosodo**
### What does this do?
Fixes duplication of entries both on cache and db
Fixes authorization by key for http-adapter and coap-adapter

### Which issue(s) does this PR fix/relate to?
NOISSUE

### List any changes that modify/break current functionality
N/A

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
No

### Notes
N/A

### CODE REVIEW

File: cmd/things/main.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func main() {
 	defer cacheClient.Close()
 
 	// Setup new auth grpc client
-	auth, authHandler, err := authClient.Setup(envPrefix, cfg.JaegerURL)
+	auth, authHandler, err := authClient.Setup(envPrefix, "localhost:6831")
```
Comment (SammyOina) [inline cmd/things/main.go:111]: why not a config from env

File: pkg/sdk/go/users_test.go [hunk 2]
```diff
@@ -470,30 +470,31 @@ func TestListMembers(t *testing.T) {
...
-			response: []sdk.User{},
+			response: []sdk.User(nil),
 			err:      errors.NewSDKErrorWithStatus(errors.ErrNotFound, http.StatusNotFound),
 		},
 	}
 
 	for _, tc := range cases {
-		repoCall := pRepo.On("Evaluate", mock.Anything, mock.Anything, mock.Anything).Return(nil)
-		repoCall1 := cRepo.On("Members", mock.Anything, mock.Anything, mock.Anything).Return(clients.MembersPage{Members: convertClients(tc.response)}, tc.err)
+		repoCall := pRepo.On("CheckAdmin", mock.Anything, mock.Anything).Return(nil)
```
Comment (SammyOina) [inline pkg/sdk/go/users_test.go]: use mock anything of type

**Review by SammyOina — CHANGES_REQUESTED** [review]

**Comment by SammyOina** [comment]
Fix failing test @rodneyosodo

**Comment by rodneyosodo** [comment]
> Fix failing test @rodneyosodo

Failing Tests are resolved by https://github.com/mainflux/mainflux/pull/1735

File: cmd/things/main.go (hunk 1 above)
Comment (rodneyosodo) [reply cmd/things/main.go:111]: This is fixed with https://github.com/mainflux/mainflux/pull/1732

File: pkg/sdk/go/users_test.go (hunk 2 above)
Comment (rodneyosodo) [reply pkg/sdk/go/users_test.go]: It accepts native types i.e `int`, `string`

**Review by SammyOina — APPROVED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**